### PR TITLE
Fix: Grid view - save new row only when all required cell values provided

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

 re #3696 

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Review by wingkwong — COMMENTED** [review]
Let's say I've column A, B, C, and D where C & B are required cells.

Case 1. Fill in A & B only. C & D are highlighted. Cell values are not saved which is expected.
Case 2. Fill in A, B, C & D. Cell values are saved which is expected.
Case 3. After Case 2, remove values from C & D. Cell values are saved. (Is it expected in this case? or the value becomes empty rather than NULL so they are considered as filled?)

**Comment by pranavxc** [comment]
> Let's say I've column A, B, C, and D where C & B are required cells.
> 
> Case 1. Fill in A & B only. C & D are highlighted. Cell values are not saved which is expected. Case 2. Fill in A, B, C & D. Cell values are saved which is expected. Case 3. After Case 2, remove values from C & D. Cell values are saved. (Is it expected in this case? or the value becomes empty rather than NULL so they are considered as filled?)


We are deciding the required field based on the database table field's `not null` attribute, if it's true need to be a non-null value and it can be an empty string as well.

File: packages/nc-gui/composables/useExpandedFormStore.ts [hunk 1]
```diff
@@ -19,7 +19,9 @@ import {
   useProvideSmartsheetRowStore,
   useSharedView,
 } from '#imports'
+import { useMetas } from '~/composables/useMetas'
 import type { Row } from '~/lib'
+import { populateInsertObject } from '~/utils'
```
Comment (bcakmakoglu) [inline packages/nc-gui/composables/useExpandedFormStore.ts:24]: -> `'#imports'` 😉

File: packages/nc-gui/composables/useViewData.ts [hunk 2]
```diff
@@ -19,6 +20,7 @@ import {
   useSmartsheetStoreOrThrow,
   useUIPermission,
 } from '#imports'
+import { useMetas } from '~/composables/useMetas'
```
Comment (bcakmakoglu) [inline packages/nc-gui/composables/useViewData.ts:23]: -> `'#imports'`

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Review by wingkwong — APPROVED** [review]

**Review by bcakmakoglu — APPROVED** [review]

**Pull request merged**